### PR TITLE
Support none defaultSettings

### PR DESCRIPTION
**Pull request by pepibumur**
### Short description 📝
Pairing with @RomainBoulay on porting a project to Swift, we figured out that there's no way to disable the generation of build settings by Tuist.

### Solution 📦
I'm adding a new `DefaultSettings` case, `.none`, to disable the default build settings generation.

### Implementation 👩‍💻👨‍💻
- [x] Add `DefaultSettings.none`.
- [x] Add tests.
- [x] Update documentation.
- [x] Update CHANGELOG.

### CODE REVIEW

**Comment by pepibumur** [comment]
@marciniwanicki / @kwridan I have the feeling there's already a way to achieve this but I couldn't find anything.

File: docs/usage/2-manifest.mdx [hunk 1]
```diff
@@ -678,9 +678,13 @@ description: "Recommented settings including warning flags to help you catch som
...
 description: "A minimal set of settings to make the project compile without any additional settings for example `PRODUCT_NAME` or `TARGETED_DEVICE_FAMILY`."
 },
+{
+case: ".none",
+description: "Tuist won't generate any build settings for the target or project."
+},
 ]}/>
 
-<Message info title="Essential Settings" description="The `.essential` option can be used in the event all warnings are defined in an `xcconfig` file to prevent Tuist from overriding them at the project or target level."/>
+<Message info title="Essential Settings" description="The .essential option can be used in the event all warnings are defined in an `xcconfig` file to prevent Tuist from overriding them at the project or target level."/>
```
Comment (kwridan) [inline docs/usage/2-manifest.mdx]: Worth adding a warning style comment for the `.none` case highlighting Tusit is delegating the responsibility of settings to the user and the project may fail to compile! :)

Comment (kwridan) [reply docs/usage/2-manifest.mdx]: How does this look?

> Projects may fail to compile if some essential settings are missing. Use `.none` only if you are specifying all the necessary build settings manually or via xcconfig files.

**Comment by kwridan** [comment]
Push an update to expose `.none`.

There are still some settings that get generated by Tuist - all the ones within https://github.com/tuist/tuist/blob/master/Sources/TuistGenerator/Generator/ConfigGenerator.swift#L156 - those are essential to Tuist's functionality (e.g. ensuring Test hosts are setup etc...) - perhaps to allow user control over some of those, we can change the order of precedence

- Set defaults
- Apply target derived settings 
- Apply custom user specified settings

**Comment by pepibumur** [comment]
> those are essential to Tuist's functionality (e.g. ensuring Test hosts are setup etc...) - perhaps to allow user control over some of those, we can change the order of precedence

Isn't that the current order? I think it makes sense to keep Tuist's derived settings because we infer those from the manifest. We could allow the user to control them but if things don't work, they'll blame Tuist for that.

File: docs/usage/2-manifest.mdx (hunk 1 above)
Comment (pepibumur) [reply docs/usage/2-manifest.mdx]: Something like this @kwridan ?

image

**Pull request merged**